File: app/src/main/java/kolar/simplealarm/Model/AlarmClass.java
package kolar.simplealarm.Model;

import android.support.annotation.IntDef;

import java.io.Serializable;
import java.lang.annotation.Retention;
import java.util.Calendar;
import java.util.GregorianCalendar;

import static java.lang.annotation.RetentionPolicy.SOURCE;

/**
 * Created by Kolar on 25.11.2017.
 */

public class AlarmClass implements Serializable {

    @IntDef({POSTPONE_MODE_NONE, POSTPONE_MODE_FIVE_MINUTES, POSTPONE_MODE_TEN_MINUTES, POSTPONE_MODE_FIFTEEN_MINUTES, POSTPONE_MODE_THIRTY_MINUTES})
    @Retention(SOURCE)
    public @interface PostponeMode {
    }

    public static final int POSTPONE_MODE_NONE = -1;
    public static final int POSTPONE_MODE_FIVE_MINUTES = 5;
    public static final int POSTPONE_MODE_TEN_MINUTES = 10;
    public static final int POSTPONE_MODE_FIFTEEN_MINUTES = 15;
    public static final int POSTPONE_MODE_THIRTY_MINUTES = 30;

    private boolean activate = false;
    private Calendar date = GregorianCalendar.getInstance();
    private boolean repeat = false;
    private long id;

    @PostponeMode
    private int postpone;

    public AlarmClass() {
        date.set(Calendar.SECOND, 0);
        this.postpone = POSTPONE_MODE_NONE;
    }

    public void setPostponeMode(@PostponeMode int mode) {
        this.postpone = mode;
    }

    @PostponeMode
    public int getPostponeMode() {
        return this.postpone;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public boolean isActivate() {
        return activate;
    }

    public void setActivate(boolean activate) {
        this.activate = activate;
    }

    public Calendar getDate() {
        return date;
    }

    public void setDate(GregorianCalendar date) {
        this.date = date;
    }

    public boolean isRepeat() {
        return repeat;
    }

    public void setRepeat(boolean repeat) {
        this.repeat = repeat;
    }

    public void setDate(Calendar date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return id + ";" + date.getTimeInMillis() + ";" + postpone + ";" + repeat + ";" + activate;
    }
}

File: app/src/main/java/kolar/simplealarm/Model/AlarmClass.kt
package kolar.simplealarm.Model

import android.support.annotation.IntDef
import java.io.Serializable
import java.util.*


/**
 * Created by ondrej.kolar on 22.01.2018.
 */
class AlarmClass : Serializable {

    @IntDef(POSTPONE_MODE_NONE, POSTPONE_MODE_FIVE_MINUTES, POSTPONE_MODE_TEN_MINUTES, POSTPONE_MODE_FIFTEEN_MINUTES, POSTPONE_MODE_THIRTY_MINUTES)
    @Retention(AnnotationRetention.SOURCE)
    annotation class PostponeMode

    companion object {
        const val POSTPONE_MODE_NONE = 0L
        const val POSTPONE_MODE_FIVE_MINUTES = 5L
        const val POSTPONE_MODE_TEN_MINUTES = 10L
        const val POSTPONE_MODE_FIFTEEN_MINUTES = 15L
        const val POSTPONE_MODE_THIRTY_MINUTES = 30L
    }

    var activate = false
    var date = GregorianCalendar.getInstance()
    var repeat = false
    var id: Long = 0
    private var postpone: Long = POSTPONE_MODE_NONE

    fun setPostpone(@PostponeMode postpone: Long) {
        this.postpone = postpone
    }

    fun getPostpone(): Long {
        return this.postpone
    }

    override fun toString(): String {
        return id.toString() + ";" + date.timeInMillis + ";" + postpone + ";" + repeat + ";" + activate
    }
}